integration-test.py

```diff
@@ -503,9 +503,17 @@ def query_result(self, query):
                          params=dict(data=query))
             if r.status_code == 200:
                 break
-            if r.status_code != 429:
+            if r.status_code != 429 and r.status_code != 504:
```

Comment:
https://github.com/tilezen/vector-datasource/pull/1336/commits/210227d40bcb6f5088163b9d68c89ebb66dd2246 is inspired by this comment. I kept 200 and 429/504 separately.